api/python/provisioner/commands/deploy_vm.py

```diff
@@ -144,12 +136,16 @@ def _run_states(self, states_group: str, run_args: run_args_type):
                     self._apply_state(
                         f"components.{state}", secondaries, stages
                     )
-                elif state in ("ha.cortx-ha"):
-                    #Execute firt on primary, then on second node and then on the third
+                elif state in (
+                    "ha.cortx-ha",
+                ):
+                    # Execute first on primary then on secondary sequentially.
+                    nodes = Deploy._get_node_list()
```

Comment:
shouldn't we worry about running consul here as well?

Reply:
same as above comment, if required we can have separate PR for consul check